### DISCUSSION TITLE
performance issue on android build

### DISCUSSION THREAD

**Discussion by Kwanghyuk-Kim** [stella-cv/stella_vslam#245] [Q&A] [answered]
I recently noticed that openvslam works with android too after reading of below issue link.
https://github.com/xdspacelab/openvslam/issues/240

And, from our side, it worked after long and boring of cross compiling effort with ndk. :)
But, unfortunately, the performance doesn't look nice with initial android build yet - looks like 2 fps on Samsung Galaxy-S20 smart phone, as @williamliuok mentioned in the above github issue.

Therefore, we have tried to profile the openvslam binary on the android target device by using simpleperf tool.
Unfortunately, the profiling output doesn't look promising to list up the victim for now, because tool just shows the symbols of system call as candidates that might cause the low performance.

Is there any good idea to profile it very well ?
or is there any good idea to make android build work faster or similar like pc build ?
What will be the nice beginning point to optimize the performance on android device ?
I am looking forward to getting your attention and feedback for this question. :)

**Comment by ymd-stella** [comment, marked as answer]
I am not familiar with android, but if a profiler is not available, I think the only way is to embed code for time measurement or measure from log timestamps.
